parseAPI/src/debug_parse.C

```diff
@@ -59,23 +57,28 @@ int Dyninst::ParseAPI::parsing_printf_int(const char *format, ...)
 
     if(!dyn_debug_parsing) return 0;
     if(NULL == format) return -1;
-    if (log_file == NULL) {
-        char filename[128];
-#if defined(_OPENMP)
-        snprintf(filename, 128, "%s-%d.txt", getenv("DYNINST_DEBUG_PARSING"), omp_get_thread_num());
-#else
-        snprintf(filename, 128, "%s-%d.txt", getenv("DYNINST_DEBUG_PARSING"), 0);
-#endif
 
-        log_file = fopen(filename, "w");
-    }
+    auto const id = []() -> int {
+#ifdef _OPENMP
+      return omp_get_thread_num();
+#endif
+      return 0;
+    }();
 
     va_list va;
     va_start(va,format);
-    int ret = vfprintf(log_file, format, va);
-    fflush(log_file);
-    va_end(va);
 
+    auto ret = [&]() {
+      int v{};
+      #pragma omp critical
+      {
+        v = fprintf(stderr, "[thread %d] ", id);
+        v &= vfprintf(stderr, format, va);
```

Comment:
The `&=` doesn't make sense.  fprintf returns the number of characters written or -1 on error.  You should record each return value separately in `r1` and `r2` with a return value of `r1 != -1 && r2 != -1) ? (r1 + r2) : -1`.  Slightly better would be to also only execute the second `vprintf` if `r1 != -1`.  It is also likely that nothing looks at the return value.